src/libraries/System.Data.Common/src/System/Data/DataSet.cs

```diff
@@ -367,6 +374,7 @@ internal void DeserializeDataSet(SerializationInfo info, StreamingContext contex
         }
 
         // Deserialize schema.
+        [RequiresUnreferencedCode(RequiresUnreferencedCodeMessage)]
```

Comment:
Same comment here, seems like the only unsafe path is BinaryFormatter.Deserialize right?

Reply:
as above (XML itself can contain filter expressions which are unsafe)